src/main/java/com/blamejared/contenttweaker/fluids/FluidBuilder.java

```diff
@@ -0,0 +1,192 @@
+
+            @Override
+            public ResourceLocation getRegistryName() {
+                return location;
+            }
+
+            @Override
+            public FlowingFluid get() {
+                return Objects.requireNonNull(fluid);
+            }
+
+            @Override
+            public FlowingFluidBlock getFluidBlock() {
+                return Objects.requireNonNull(fluidBlock);
+            }
+
+            @Override
+            public void updateFluid(FlowingFluid fluid) {
+                this.fluid = (ForgeFlowingFluid.Source) fluid;
+            }
+
+            @Override
+            public void setFluidBlock(FlowingFluidBlock fluidBlock) {
+                this.fluidBlock = fluidBlock;
+            }
+        };
+        IIsCotFluid flowingFluid = new IIsCotFluid() {
+            private ForgeFlowingFluid.Flowing fluid;
+
+            @Override
+            public ResourceLocation getRegistryName() {
+                return new ResourceLocation(ContentTweaker.MOD_ID, path + "_flowing");
+            }
+
+            @Override
+            public FlowingFluid get() {
+                return Objects.requireNonNull(fluid);
+            }
+
+            @Nullable
+            @Override
+            public FlowingFluidBlock getFluidBlock() {
+                return null;
+            }
+
+            @Override
+            public void updateFluid(FlowingFluid fluid) {
+                this.fluid = (ForgeFlowingFluid.Flowing) fluid;
+            }
+        };
+        ForgeFlowingFluid.Properties properties = new ForgeFlowingFluid.Properties(stillFluid, flowingFluid, builder);
+        AbstractBlock.Properties fluidBlockProperties = AbstractBlock.Properties.create(isMolten ? Material.WATER : Material.LAVA)
+                .doesNotBlockMovement()
+                .hardnessAndResistance(100.0f)
+                .noDrops();
+        FlowingFluidBlock fluidBlock = new FlowingFluidBlock(stillFluid, fluidBlockProperties);
+        CoTFluidBucketItem bucketItem = new CoTFluidBucketItem(stillFluid, new Item.Properties().maxStackSize(1).containerItem(Items.BUCKET).group(ItemGroup.MISC));
+        bucketItem.setRegistryName(path + "_bucket");
+        bucketItem.setItemColorSupplier(((stack, tintIndex) -> tintIndex == 1 ? color : -1));
+        fluidBlock.setRegistryName(stillFluid.getRegistryNameNonNull());
+        properties.block(() -> fluidBlock);
+        properties.bucket(() -> bucketItem);
+        stillFluid.setFluidBlock(fluidBlock);
+        flowingFluid.updateFluid((FlowingFluid) new ForgeFlowingFluid.Flowing(properties).setRegistryName(flowingFluid.getRegistryNameNonNull()));
+        stillFluid.updateFluid((FlowingFluid) new ForgeFlowingFluid.Source(properties).setRegistryName(stillFluid.getRegistryNameNonNull()));
+        VanillaFactory.queueFluidForRegistration(stillFluid);
+        VanillaFactory.queueFluidForRegistration(flowingFluid);
+        VanillaFactory.queueItemForRegistration(bucketItem);
```

Comment:
Maybe some empty lines or methods to separate the different things this block of code does, to make it easier to read?